lxd/instance/drivers/driver_qemu.go

```diff
@@ -3051,6 +3051,33 @@ func (d *qemu) addNetDevConfig(cpuCount int, busName string, qemuDev map[string]
 			pciSlotName = nicItem.Value
 		} else if nicItem.Key == "pciIOMMUGroup" {
 			pciIOMMUGroup = nicItem.Value
+		} else if nicItem.Key == "mtu" {
+			mtuStr = nicItem.Value
+		}
+	}
+
+	if d.localConfig["agent.rename_interfaces"] != "" {
+		mtuInt, err := strconv.Atoi(mtuStr)
+		if err != nil {
+			logger.Warnf("Invalid MTU returned by device %q run configuration", devName)
+			mtuInt = 0
+		}
+		mtu := uint32(mtuInt)
+
+		nicConfigBytes, err := json.Marshal(deviceConfig.NICConfig{Name: devName, MTU: mtu})
+		if err != nil {
+			return nil, err
+		}
+
+		nicConfigDir := filepath.Join(d.Path(), "config", "nics")
+		err = os.MkdirAll(nicConfigDir, 0500)
```

Comment:
As `generateConfigShare` runs first and already creates the "config" dir, I think we could get it to create the "nics" dir too, and then we can just expect it to exist and create the file, saves doing multiple syscalls for each device, and also avoids silently creating the dirs if somehow the ordering is changed/broken in the future.

Reply:
Oh also we should ensure in generateConfigShare that if the `nics` dir exists already, it is cleared out/removed & recreated, so we can ensure any old/renamed NICs aren't left in there.